### PR TITLE
[Worker] Fix will not kill the subprocess in remote when stop a remote-shell task #15570

### PR DESCRIPTION
**Pull request by shilei-ly**
## Purpose of the pull request

fix cannot kill the subprocess when stop a remote-shell task #15570
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-task-plugin/dolphinscheduler-task-remoteshell/src/main/java/org/apache/dolphinscheduler/plugin/task/remoteshell/RemoteExecutor.java [hunk 1]
```diff
@@ -154,11 +162,58 @@ public void cleanData(String taskId) {
...
+        log.info("prepare to parse pid, raw pid string: {}", rawPidStr);
+        ArrayList<String> allPidList = new ArrayList<>();
+        Matcher mat = pattern.matcher(rawPidStr);
+        if (null != mat) {
+            while (mat.find()) {
+                allPidList.add(mat.group(1));
+            }
+        }
+        return String.join(" ", allPidList).trim();
+    }
```
Comment (ruanwenjun) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-remoteshell/src/main/java/org/apache/dolphinscheduler/plugin/task/remoteshell/RemoteExecutor.java]: it's better to move this into `ProcessUtils` ?

Comment (shilei-ly) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-remoteshell/src/main/java/org/apache/dolphinscheduler/plugin/task/remoteshell/RemoteExecutor.java]: @ruanwenjun  Hi，I moved this logic into processutils

File: dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/utils/ProcessUtils.java [hunk 2]
```diff
@@ -146,6 +147,32 @@ public static String getPidsStr(int processId) throws Exception {
         return sb.toString().trim();
     }
 
+    public static String parsePidStr(String rawPidStr) {
```
Comment (rickchengx) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/utils/ProcessUtils.java:136]: This function has some common logic with function `getPidsStr`. I think the common logic should be extracted.

**Review by rickchengx — APPROVED** [review]
LGTM, wait for CI

**Pull request merged**